components/unit/UnitVegaViz.vue

```diff
@@ -13,22 +13,27 @@
         </VCol>
       </VRow>
     </template>
-    <template v-else>
-      <VRow>
-        <VCol>
-          <i>data in this format cannot be displayed by this visualization</i>
-        </VCol>
-      </VRow>
-    </template>
+    <i v-else-if="isValid">No data found</i>
+    <i v-else>Data in this format cannot be displayed by this visualization</i>
   </div>
 </template>
 <script>
 import embed from 'vega-embed'
+import _ from 'lodash'
 
 function isDataValid(data) {
-  return !data || !!(data.items?.length > 0)
+  return (
+    _.every(
+      ['items', 'headers'],
+      field => _.has(data, field) && Array.isArray(data[field])
+    ) &&
+    data.headers.length > 0 &&
+    _.every(data.items, i => _.every(data.headers, h => _.has(i, h)))
+  )
+}
```

Comment:
Another style comment: Multiline boolean expressions are harder to read than if statements. I would try to do something like this

```
const {headers, items} = data
if (!Array.isArray(headers) && !Array.isArray(items) ) {
  return false
}
if (headers.length === 0) {
  return false
}
const missingHeader = items.find(item => item.length < headers.length)
if(missingHeader){
  return false
}
```
(Also I try to avoid lodash if the standard library can do the job without too many complications)

Reply:
I don't necessarily agree that multiline boolean expressions are harder to read.
Also, I think that lodash's functional and compact style allows writing code that is less error-prone.